specs/conductor.md

```diff
@@ -1,59 +1,137 @@
+
+The architecture of the Conductor is inspired by the [Actor Model](https://en.wikipedia.org/wiki/Actor_model) with the actors within the Conductor being the `Driver`, `Reader`, and `Executor`. Each actor operates concurrently and communicates with the other actors by passing messages. The Conductor is written in Rust and utilizes the Tokio runtime to achieve this.
 
 ![Conductor Architecture](assets/conductor-architecture.png)
 
-The responsibilities of each module are as follows:
 ### Driver
- - Top level coordinator that runs and manages all the sub-components necessary for the Conductor
- - Creates the `Reader` and `Executor` actors on startup
- - Creates a gossip network for receiving data from the sequencer network
-     - The gossip network uses the [astria-gossipnet](https://github.com/astriaorg/astria/tree/main/crates/astria-gossipnet)
- - Runs an event loop that handles receiving `DriverCommand`s and messages from the gossip network
- - Passes sequencer blocks received from the gossip network to the `Executor`
+
+- Top level coordinator that runs and manages all the subcomponents necessary for the Conductor
+- Creates the `Reader` and `Executor` actors on startup
+- Creates a gossip network for receiving data from the Sequencer network
+    - The gossip network uses the [astria-gossipnet](https://github.com/astriaorg/astria/tree/main/crates/astria-gossipnet)
+- Runs an event loop that handles receiving `DriverCommand`s from other actors as well as messages from the gossip network
+- Validates and passes Sequencer blocks received from the gossip network to the `Executor`
+
+The Driver receives either `Events` ([link](https://github.com/astriaorg/astria/blob/6e71a76fa52c522ffdcabcd9d659e4de765d9d61/crates/astria-gossipnet/src/network_stream.rs#L39)) from the network, or `DriverCommand`s ([link](https://github.com/astriaorg/astria/blob/6e71a76fa52c522ffdcabcd9d659e4de765d9d61/crates/astria-conductor/src/driver.rs#L54)) from
+the Conductor's internal event loop on a timer. The variants for `Event` and
+`DriverCommand` that are relevant to the processing of blocks within the
+Conductor are:
+
+- `Event::GossipsubMessage(Message)` ([link](https://github.com/astriaorg/astria/blob/6e71a76fa52c522ffdcabcd9d659e4de765d9d61/crates/astria-gossipnet/src/network_stream.rs#L50))
+    - The `Message` value within the `GossipsubMessage` contains the Sequencer
+      block data from the Astria Sequencer. The `Message` is then parsed to a
+      `SequencerBlockData`
+      ([link](https://github.com/astriaorg/astria/blob/6e71a76fa52c522ffdcabcd9d659e4de765d9d61/crates/astria-sequencer-types/src/sequencer_block_data.rs#L39)).
+      Once transformed, the block data is validated to make sure that the
+      validator set correctly agreed on the block, the proposer for the Sequencer
```

Comment:
```suggestion
      Once transformed, the block data is validated to make sure that the proposer for the Sequencer
```
there isn't any commit/proposer set validation on the gossipnet block as it's the head-of-chain block which isn't committed/finalized yet